### ISSUE TITLE
Does it work on Windows ?

### ISSUE THREAD

**Issue by mariusmg** [orakaro/rainbowstream#55]
Win8.1, Python 2.7.8

When running it , i get the following error

Traceback (most recent call last):
  File "C:\python2\Scripts\rainbowstream-script.py", line 5, in <module>
    from pkg_resources import load_entry_point
  File "C:\python2\lib\site-packages\setuptools-0.6c11-py2.7.egg\pkg_resources.py", line 2603, in <module>
  File "C:\python2\lib\site-packages\setuptools-0.6c11-py2.7.egg\pkg_resources.py", line 666, in require
  File "C:\python2\lib\site-packages\setuptools-0.6c11-py2.7.egg\pkg_resources.py", line 565, in resolve
DistributionNotFound: Pillow

If i try to manually install pillow i get

pip install pillow
Requirement already satisfied (use --upgrade to upgrade): pillow in c:\python2\lib\site-packages
Cleaning up...

**Comment by nkvenom** [comment]
Yo can install it using the conda python distribution, using:

> conda install pillow
Or downloading the library from for your version of python:
http://www.lfd.uci.edu/~gohlke/pythonlibs/#pillow

But the problem I think is with a file called image.c that can get compiled on windows

**Comment by iainsgillis** [comment]
OS: Windows 8.1
python --version 2.7.10 [virtual environment twt2]
python --version 3.4.3 [virtual environment twt3]

---

@nkvenom Indeed pillow does install no problem with conda.
@DTVD cool project. Nice work. Running on my manjaro tinkering machine with no trouble at all.

For us folks who use Windows as our primary machines, I'm keen to try to get this working. Alas, it's beyond my skill. Any advice/guidance would be welcome.

---

#### python 2.7.10
Traceback (most recent call last):
  File "C:\python\envs\twt2\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\python\envs\twt2\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\python\envs\twt2\Scripts\rainbowstream.exe\__main__.py", line 5, in <module>
  File "C:\python\envs\twt2\lib\site-packages\rainbowstream\rainbow.py", line 22, in <module>
    from .draw import *
  File "C:\python\envs\twt2\lib\site-packages\rainbowstream\draw.py", line 15, in <module>
    from .c_image import *
  File "C:\python\envs\twt2\lib\site-packages\rainbowstream\c_image.py", line 25, in <module>
    rgb2short = call_c()
  File "C:\python\envs\twt2\lib\site-packages\rainbowstream\c_image.py", line 21, in call_c
    image_c = ctypes.cdll.LoadLibrary(library)
  File "C:\python\envs\twt2\lib\ctypes\__init__.py", line 444, in LoadLibrary
    return self._dlltype(name)
  File "C:\python\envs\twt2\lib\ctypes\__init__.py", line 366, in __init__
    self._handle = _dlopen(self._name, mode)
WindowsError: [Error 126] The specified module could not be found

---

#### python 3.4.3

> **NB**: to install `rainbowstream` with python3, I had to follow the instructions  https://github.com/DTVD/rainbowstream/blob/master/README.rst#development, and changed line 31 on rainbowstream\setup.py to<br/>`long_description=open("./README.rst", "r", encoding="utf-8").read(),`<br/>in order to fix the UnicodeDecodeError.

Traceback (most recent call last):
  File "C:\python\envs\twt3\Scripts\rainbowstream-script.py", line 9, in <module>
    load_entry_point('rainbowstream==1.3.1', 'console_scripts', 'rainbowstream')()
  File "C:\python\envs\twt3\lib\site-packages\setuptools-18.1-py3.4.egg\pkg_resources\__init__.py", line 558, in load_entry_point
  File "C:\python\envs\twt3\lib\site-packages\setuptools-18.1-py3.4.egg\pkg_resources\__init__.py", line 2682, in load_entry_point
  File "C:\python\envs\twt3\lib\site-packages\setuptools-18.1-py3.4.egg\pkg_resources\__init__.py", line 2355, in load
  File "C:\python\envs\twt3\lib\site-packages\setuptools-18.1-py3.4.egg\pkg_resources\__init__.py", line 2361, in resolve
  File "c:\python\envs\rainbowstream\rainbowstream\rainbow.py", line 22, in <module>
    from .draw import *
  File "c:\python\envs\rainbowstream\rainbowstream\draw.py", line 15, in <module>
    from .c_image import *
  File "c:\python\envs\rainbowstream\rainbowstream\c_image.py", line 25, in <module>
    rgb2short = call_c()
  File "c:\python\envs\rainbowstream\rainbowstream\c_image.py", line 21, in call_c
    image_c = ctypes.cdll.LoadLibrary(library)
  File "C:\python\envs\twt3\lib\ctypes\__init__.py", line 429, in LoadLibrary
    return self._dlltype(name)
  File "C:\python\envs\twt3\lib\ctypes\__init__.py", line 351, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: [WinError 126] The specified module could not be found

**Comment by DTVD** [comment]
@iainsgillis great new that Pillow got installed fine in Window :+1: . This `image.so` problem maybe easier to deal with, since I'm using a Unix-specific compiler in this line:
https://github.com/DTVD/rainbowstream/blob/master/rainbowstream/c_image.py#L19

Put an Windown specific in here will finally solve all the problem (perhaps). Sorry that I'm not have an environment to test, so a PR from a Window user may help a lots :)

**Comment by iainsgillis** [comment]
@DTVD Sorry for the slow-coming reply -- busy week!

@mariusmg
I installed [babun](https://github.com/babun/babun/), which seems to have fixed the compilation problem. I've forked your project and made a couple of changes to get it running on Windows. The 'h' function isn't working (producing an `OMG something is wrong with Twitter API right now.` error), but everything else is just fine. :) 

NB: I've only tested this on Python 2.7.10, 32-bit.
